scrapy/aws.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import smtplib
+from email import encoders as Encoders
+from email.mime.base import MIMEBase
+from email.mime.multipart import MIMEMultipart
+from email.mime.nonmultipart import MIMENonMultipart
+from email.mime.text import MIMEText
+from email.utils import COMMASPACE, formatdate
+from io import BytesIO
+
+from scrapy.utils.misc import arg_to_iter
+
+class SESender(object): 
+
+    def __init__(self, AWS_HOST_NAME, AWS_USER_NAME, AWS_PASSWORD, AWS_PORT_NUMBER, FROM_ADDRESS):
+        self.host_name = AWS_HOST_NAME
+        self.user_name = AWS_USER_NAME
+        self.password = AWS_PASSWORD
+        self.port_number = AWS_PORT_NUMBER
+        self.from_address = FROM_ADDRESS
+
+    def construct_message(self, to, subject, body, cc, attachments, mimetype, charset):
+
+        # Check to see whether attachments exist
+        if attachments: 
+            message = MIMEMultipart()
+        else: 
+            message = MIMENonMultipart(*mimetype.split('/', 1))
+
+        to = list(arg_to_iter(to))
+
+        # Construct the message data-structure
+        message = MIMEMultipart()
+
+        # Add content to the body
+        message['From'] = self.from_address
+        message['To'] = COMMASPACE.join(to)
+        message['Date'] = formatdate(localtime = True)
+        message['Subject'] = subject
+        message['Body'] = body
+        recipients = to[:]
```

Comment:
`to.copy()` looks more readable to me, but I guess that's just a matter of personal preference.